plutus-core/src/Language/PlutusCore/Parser.y

```diff
@@ -113,10 +111,9 @@ Term : Var                                                 { $1 }
      | openBrace   Term some(Type)           closeBrace    { tyInst $1 $2 (NE.reverse $3) }
      | openParen   lam Name Type Term        closeParen    { LamAbs $2 $3 $4 $5 }
      | openBracket Term some(Term)           closeBracket  { app $1 $2 (NE.reverse $3) }
-     | openParen   con builtintypeid literal closeParen    { % mkBuiltinConstant (tkLoc $3) -- % = monadic action
-	                                                       (tkBuiltinTypeId $3) (tkLoc $4) (tkLiteralConst $4)}
+     | openParen   con builtintypeid literal closeParen    { % fmap (uncurry Constant) (mkBuiltinConstant (tkLoc $3) (tkBuiltinTypeId $3) (tkLoc $4) (tkLiteralConst $4)) }
```

Comment:
So we can share `makeBuiltin` between typed and untyped. They both have a `Builtin` constructor in their respective different `Term`s: we can't return `Term`s and share it, because they're different `Term`s, so we return the arguments of their respective `Builtin` constructors. I'll add a note.

Reply:
Ah, of course.  Thanks!